### PR TITLE
Reading list

### PR DESCRIPTION
**Pull request by daytonn**

### CODE REVIEW

File: elements/bulbs-reading-list/lib/reading-list-article.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      this.pushToHistory();
+    }
+  }
+
+  startedReading (oldProgress, newProgress) {
+    return (oldProgress === 0 && newProgress > 0) || (oldProgress === 100 && newProgress < 100)
+  }
+
+  pushToHistory () {
+    window.history.pushState(null, this.title, this.href);
```
Comment (spra85) [inline elements/bulbs-reading-list/lib/reading-list-article.js]: This is where we would hook in page view analytics

File: elements/bulbs-reading-list/lib/reading-list-article.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      percentage = position / height * 100;
+    }
+    const progress = Math.abs(Math.floor(percentage));
+
+    return progress > 100 ? 100 : progress;
+  }
+
+  pushStateIfStartedReading (oldProgress, newProgress) {
+    if (this.startedReading(oldProgress, newProgress)) {
+      this.pushToHistory();
```
Comment (spra85) [inline elements/bulbs-reading-list/lib/reading-list-article.js]: This is where we would hook in page view analytics

**Comment by MelizzaP** [comment]
[handle pr review notes](https://trello.com/c/KpEfgG4b/74-handle-pr-review-notes)

File: elements/bulbs-reading-list/lib/reading-list-article.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      percentage = position / height * 100;
+    }
+    const progress = Math.abs(Math.floor(percentage));
+
+    return progress > 100 ? 100 : progress;
+  }
+
+  prepareAnalytics () {
+    let targeting = this.dimensions;
+    // need to update these to match correct implementation - scook
```
Comment (collin) [inline elements/bulbs-reading-list/lib/reading-list-article.js]: Does this still need to happen?

Comment (shawncook) [inline elements/bulbs-reading-list/lib/reading-list-article.js]: I think @spra85 helped out with this @collin

Comment (spra85) [inline elements/bulbs-reading-list/lib/reading-list-article.js]: @collin, we're going to tackle post-launch.

**Pull request merged**